### DISCUSSION TITLE
Ability to specify path separator

### DISCUSSION THREAD

**Discussion by elijaholmos** [101arrowz/fflate#147] [Q&A] [answered]
I'm using this library on my local Windows 10 device to programmatically zip up a directory of browser extension source files; a pretty simple operation. I ran into an interesting problem the other day: the browser store I was submitting my extension to did not support backslashes as the path separator for directories in zip files. A maintainer of the extension store's validation software confirmed that this was intentional, and advised that I either switch to a different zipping library, or find a way to implement the forwardslash separator with `fflate`. I briefly perused the documentation, but couldn't find anything that mentioned such an option.

Is this behavior currently possible with `fflate`? If not, may I submit a feature request?

Related: https://github.com/mozilla/addons-server/issues/19762

**Comment by 101arrowz** [comment, marked as answer]
`fflate` always adds forward slashes in ZIP files unless you manually use paths with backward slashes. Could you check if that's what's happening?

**Reply by 101arrowz** [reply]
Just checked your code, you might want to use the object syntax instead of files to zip.

```js
import { zipSync } from 'fflate';

zipSync({
  folder1:  {
    folder2: {
      // folder1/folder2/out.txt
      'out.txt': new Uint8Array([97])
    }
  }
})
```

**Reply by elijaholmos** [reply]
that works, thank you. the filename approach didn't work because I was manually specifying paths?

**Reply by 101arrowz** [reply]
Yeah, your `path.relative` call was returning a Windows path. `fflate` will take your filepath and add it to the ZIP, no questions asked; it won't convert it to Unix for you. You could possibly also have used `path.posix.relative` or something but using this directory structure is better because it also creates directory entries in the ZIP.

**Reply by elijaholmos** [reply]
I see. thanks again for your help & explanation
